core/src/main/java/com/graphhopper/routing/Router.java

```diff
@@ -561,6 +561,9 @@ protected FlexiblePathCalculator createPathCalculator(QueryGraph queryGraph) {
                 throw new IllegalArgumentException("Cannot find LM preparation for the requested profile: '" + profile.getName() + "'" +
                         "\nYou can try disabling LM using " + Parameters.Landmark.DISABLE + "=true" +
                         "\navailable LM profiles: " + landmarks.keySet());
+            if (profile instanceof CustomProfile && request.getCustomModel() != null
+                    && !request.getHints().getBool("lm.disable", false))
```

Comment:
The `lm.disable` check is not needed here right? It is already done before when we select the `LMSolver`.